scripts/migrations.js

```diff
@@ -0,0 +1,79 @@
+import Constants from '../constants.js';
+import Settings from '../settings.js';
+
+export default class Migrations {
+	constructor() {
+		this._settings = new Settings();
+	}
+	/* Migrates the description of Custom Convenient Effects to use the ActiveEffect5e#description field instead of the legacy flags.
+	 *
+	 * @param {string} id - the ID of the custom effects item
+	 * @returns {Promise} a promise that resolves when all the Item's ActiveEffects descriptions have migrated
+	 */
+	async _migrateCustomEffectsItemEffectDescriptions(id, migration) {
+		const item = this._settings._findCustomEffectsItem(id);
+		const updates = [];
+		const effects = item.effects.filter((effect) =>
+			effect.getFlag(Constants.MODULE_ID, Constants.FLAGS.HAS_MIGRATED)
+		);
+		for (const effect of effects) {
```

Comment:
You could replace this entire for loop with a map after the filter and also get rid of the updates empty array initialization.

```
const updates = item.effects
  .filter( ...)
  .map(effect => {
    // the logic...
    return {} // return what you would otherwise pass into `updates.push`
  })